doc/nrf/migration/migration_guide_1.x_to_2.x.rst

```diff
@@ -0,0 +1,430 @@
+.. _ncs_2.0.0_migration:
+
+Migration notes for |NCS| v2.0.0
+################################
+
+.. contents::
+   :local:
+   :depth: 2
+
+In version 2.0.0, the |NCS| introduced a number of breaking changes into its codebase.
+If your application was built using |NCS| v1.x.x, make sure to complete the actions listed in the following sections to migrate your application to |NCS| v2.x.x.
+
+Toolchain change
+****************
+
+|NCS| no longer uses the GNU Arm Embedded Toolchain by default.
+The current default toolchain is the `Zephyr SDK`_ instead, which now supports all three main operating systems (Microsoft Windows, macOS, and GNU/Linux).
+
+*Required action*: If you do not use the Toolchain Manager, install the Zephyr SDK by following the steps in "Install a Toolchain" in Zephyr's :ref:`zephyr:getting_started`.
+The updated Toolchain Manager already contains it.
+
+|VSC| extension IDE is now the default
+**************************************
+
+|VSC| extension IDE has replaced SEGGER Embedded Studio as the default supported IDE for working with the |NCS|.
+
+*Required action*: If you are building the application or sample using SEGGER Embedded Studio or on the command line and want to migrate to |VSC| extension IDE, follow the instructions in the `migrating from other IDEs to VS Code <Migrating IDE_>`_ documentation.
+
+Pin control transition
+**********************
+
+Zephyr v3.x.x introduces new state-based pin control API (`pinctrl`).
+For more details, see :ref:`zephyr:pinctrl-guide` in Zephyr documentation.
+
+*Required action*: The :ref:`ug_pinctrl` user guide provides information on how to migrate your application to use the new pin control mechanism.
+
+TF-M replaces SPM as the default secure firmware solution
+*********************************************************
+
+Trusted Firmware M (TF-M) replaces the Secure Partition Manager (SPM) for secure image firmware.
+TF-M is now enabled by default for most nRF9160 and nRF5340 applications and samples.
+
+
+*Required action*: To start using TF-M instead of SPM in your application, see the information in :ref:`ug_tfm_migrate`.
```

Comment:
:ref:`ug_tfm_migrate` not linking presently.

Reply:
I will rebase

Reply:
If we are planning to link to https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/ug_tfm.html?highlight=migrating#migrating-from-secure-partition-manager-to-trusted-firmware-m, its missing a label there..

Reply:
added, for some reason I thought it exists on main (probably planned to add it myself)